seismostats/catalogs/catalog.py

```diff
@@ -213,6 +224,99 @@ def from_dict(cls,
 
         return df
 
+    @classmethod
+    def from_openquake(cls, oq_catalogue: OQCatalogue,
+                       keep_time_cols=False) -> Catalog:
+        """
+        Create a (seismostats) Catalog from an openquake Catalogue.
+        The optional dependency group openquake is required for this method.
+
+        Args:
+            oq_catalogue:       The openquake catalogue.
+            keep_time_cols:     Whether the time columns: 'year', 'month',
+                                'day', 'hour', 'minute', 'second'
+                                should be kept (they are converted to 'time').
+        Returns:
+            Catalog
+
+        Raises:
+            ValueError:         If the Catalog is empty.
+        """
+        if not _openquake_available:
+            raise ImportError("the optional openquake package is not available")
+        pd_time_columns = ['year', 'month', 'day', 'hour',
+                           'minute', 'second', 'microsecond']
+
+        def _convert_to_datetime(row):
+            return datetime(row.year,
+                            row.month,
+                            row.day,
+                            row.hour,
+                            row.minute,
+                            row.second)
+        data = oq_catalogue.data
+        clone = copy.deepcopy(data)
+        length = oq_catalogue.get_number_events()
+        if length == 0:
+            raise ValueError("Conversion for an empty Catalog is not supported")
+        cat = cls(
+            {k: v for k, v in clone.items() if len(v) > 0 or length == 0})
```

Comment:
hier komme ich nicht ganz nach. zum einen hast du einen raise wenn length == 0, zum anderen instanzierst du die klasse nur wenn die len(v)>0 oder length==0, wobei letzteres ja nicht möglich ist?

Zudem, wäre es möglich die conversion für leere Kataloge zu erlauben? In automatisierten workflows gibts immer mal wieder den Fall, dass ein Katalog leer ist. dann wäre es gut dies woanderst zu handeln als bei einer conversion. Bzw. möglicherweise auch gar nicht zu handeln, weil ein leerer Katalog nichts anderes bedeutet als "keine Erdbeben", was valide ist.

Reply:
Das kommt daher, da es Probleme gab mit
```python
empty_cat = OQCatalog.make_from_dict({})
# catalogue.py:304)     self.end_year = np.max(self.data["year"])
# KeyError: 'year'
```
Das überflüssige `length == 0` ist vergessen gegangen.


Aber wir können einfach `OQCatalog()` zurückgeben im "keine Erbeben" fall.